crates/pipeline-types/src/config.rs

```diff
@@ -74,6 +72,16 @@ pub struct PipelineConfig {
     pub outputs: BTreeMap<Cow<'static, str>, OutputEndpointConfig>,
 }
 
+impl PipelineConfig {
+    pub fn from_yaml(s: &str) -> Self {
```

Comment:
doesn't need to be fixed now but maybe we can make an issue to settle on one config format, e.g., keep json everywhere or switch everything to a human friendly format, but ideally not have both

Reply:
Yes, currently it uses YAML in the database but JSON for the API, I'm in favor of the latter I think

Reply:
Will make a note of it, but likely not this PR